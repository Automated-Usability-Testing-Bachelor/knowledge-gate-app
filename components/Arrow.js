import React from "react";
import { StyleSheet, TouchableOpacity, View, Image } from "react-native";
import Colors from "../constants/Colors";

const Arrow = ({ onPress, backgroundColor, isRight }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={onPress}
        style={[styles.touchable]}
        activeOpacity={0.6}
      >
        <View
          style={[
            styles.button,
            {
              backgroundColor: backgroundColor,
            },
          ]}
        >
          <Image
            source={
              isRight
                ? require("./onboarding/right_arrow.png")
                : require("./onboarding/left_arrow.png")
            }
            style={{ height: 24, width: 24 }}
          />
        </View>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
  },
  button: {
    paddingVertical: 5,
    paddingHorizontal: 5,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 30,
    alignSelf: "center",
    width: "auto",
    borderColor: "rgba(0,27,114, 0.5)",
    borderStyle: "solid",
    borderWidth: 0.5,
  },
  touchable: {
    padding: 15,
    borderRadius: 10,
  },
  text: {
    color: "#FFF",
    lineHeight: 18,
  },
});

export default Arrow;
