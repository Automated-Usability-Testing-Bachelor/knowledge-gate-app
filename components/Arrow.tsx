import React from 'react'
import {
  GestureResponderEvent,
  StyleSheet,
  TouchableOpacity,
  View
} from 'react-native'
import { ArrowLeftIcon, ArrowRightIcon } from 'react-native-heroicons/solid'
import Colors from '../constants/Colors'

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center'
  },
  button: {
    paddingVertical: 5,
    paddingHorizontal: 5,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
    alignSelf: 'center',
    width: 'auto',
    borderColor: 'rgba(0,27,114, 0.5)',
    borderStyle: 'solid',
    borderWidth: 0.5
  },
  touchable: {
    padding: 15,
    borderRadius: 10
  },
  text: {
    color: '#FFF',
    lineHeight: 18
  }
})

type ArrowProps = {
  onPress: (event: GestureResponderEvent) => void
  backgroundColor: string
  isRight: boolean // true if right arrow, false if left arrow
}

const Arrow: React.FC<ArrowProps> = ({ onPress, backgroundColor, isRight }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={onPress}
        style={[styles.touchable]}
        activeOpacity={0.6}
      >
        <View
          style={[
            styles.button,
            {
              backgroundColor
            }
          ]}
        >
          {isRight ? (
            <ArrowRightIcon size={24} fill={Colors.red.color} />
          ) : (
            <ArrowLeftIcon size={24} fill={Colors.red.color} />
          )}
        </View>
      </TouchableOpacity>
    </View>
  )
}

export default Arrow
