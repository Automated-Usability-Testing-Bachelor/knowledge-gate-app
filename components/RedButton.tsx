import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Colors from '../constants/Colors'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center'
  },
  button: {
    backgroundColor: Colors.red.color,
    paddingHorizontal: 10,
    paddingVertical: 10,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    alignSelf: 'center'
  },
  touchable: {
    borderRadius: 10,
    alignSelf: 'center',
    padding: 12
  },
  text: {
    color: '#FFF',
    lineHeight: 18,
    paddingHorizontal: 5
  }
})

export type Props = {
  name: string
  icon?: React.ReactElement
  onPress(): void
  testID?: string
}

const RedButton: React.FC<Props> = ({ name, icon, onPress, testID }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={onPress}
        style={styles.touchable}
        activeOpacity={0.6}
        testID={testID}
      >
        <View style={styles.button}>
          <Text style={styles.text}>{name}</Text>
          {icon || <View />}
        </View>
      </TouchableOpacity>
    </View>
  )
}

export default RedButton
