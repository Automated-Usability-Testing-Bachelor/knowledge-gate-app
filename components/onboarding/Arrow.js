import React from "react";
import {StyleSheet, TouchableOpacity, View} from "react-native";
import Colors from "../../constants/Colors";
import {ArrowLeftIcon, ArrowRightIcon} from "react-native-heroicons/solid";

const Arrow = ({onPress, isRight}) => {
    return (
        <View style={styles.container}>
            <TouchableOpacity
                onPress={onPress}
                style={[styles.touchable]}
                activeOpacity={0.6}
            >
                <View
                    style={[
                        styles.button,
                        {
                            backgroundColor: isRight ? Colors.blue.color : "#FFFFFF",
                        },
                    ]}
                >
                    {
                        isRight
                            ? <ArrowRightIcon size={24} fill={Colors.offWhite.color}/>
                            : <ArrowLeftIcon size={24} fill={Colors.blue.color}/>
                    }
                </View>
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        justifyContent: "center",
    },
    button: {
        paddingVertical: 5,
        paddingHorizontal: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 30,
        alignSelf: "center",
        width: "auto",
        borderColor: "rgba(0,27,114, 0.5)",
        borderStyle: "solid",
        borderWidth: 0.5,
    },
    touchable: {
        padding: 15,
        borderRadius: 10,
    },
    text: {
        color: "#FFF",
        lineHeight: 18,
    },
});

export default Arrow;
