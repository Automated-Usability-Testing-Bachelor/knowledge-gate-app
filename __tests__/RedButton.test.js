/*eslint-disable */

import React from 'react'
import renderer from 'react-test-renderer'
import RedButton from "../components/RedButton"
import Arrow from "../components/Arrow"
import { fireEvent, render, waitFor, screen } from '@testing-library/react-native';


// eslint-disable-next-line no-undef
describe('<RedButton />', () => {
  
    it('has 1 child', () => {
        const tree = renderer.create(<RedButton />).toJSON();
        expect(tree.children.length).toBe(1)
    });


    it('text props is passed to <text /> component', () => {
        const propText = 'test text'
        const compTree = renderer.create(<RedButton name={propText} />).toJSON();
        
        const viewComp = compTree.children[0]
        const secViewComp = viewComp.children[0]
        const textComp = secViewComp.children[0]
        const textPropActual = textComp.children[0]

        expect(compTree.children[0].type).toBe('View')
        expect(viewComp.children[0].type).toBe('View')
        expect(secViewComp.children.length).toBe(2)
        expect(textComp.type).toBe('Text')
        expect(textPropActual).toBe(propText)

        
    })
    it('icon props is passed', () => {
        const propText = 'test text'
        const compTree = renderer.create(<RedButton icon={<Arrow />} />).toJSON();
        
        const viewComp = compTree.children[0]
        const secViewComp = viewComp.children[0]
        const IconComp = secViewComp.children[1]
        expect(compTree.children[0].type).toBe('View')
        expect(viewComp.children[0].type).toBe('View')
        expect(secViewComp.children.length).toBe(2)
        expect(IconComp.children.length).toBeGreaterThan(0)

        
    })

    it('Test onPress event', async () => {
        const accessibilityLabel = 'This is a test button';
        const onPress = jest.fn(()=> "pressed");
        const activeButton = render(
        <RedButton onPress={onPress} testID="button" accessibilityLabel={accessibilityLabel}/>
        )
        const { getByTestId } = activeButton;
        const button = getByTestId('button');
        expect(activeButton).toBeDefined();

        const activeButtonAsJson = activeButton.toJSON();
        expect(activeButtonAsJson).toMatchSnapshot();
        fireEvent.press(button)
        await waitFor(() => {
            expect(onPress.mock.calls.length).toBe(1)
            expect(onPress.mock.results[0].value).toBe("pressed")
        });
      
    });
    
  
});

