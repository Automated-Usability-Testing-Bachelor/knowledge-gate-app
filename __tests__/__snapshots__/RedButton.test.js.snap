// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RedButton /> Test onPress event 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "center",
        "borderRadius": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#DD2638",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "lineHeight": 18,
            "paddingHorizontal": 5,
          }
        }
      />
      <View />
    </View>
  </View>
</View>
`;
